src/components/StatusUpdater/index.tsx

```diff
@@ -7,23 +7,54 @@ import { UserInfo } from '@respond/types/userInfo';
-]
-const optionTexts = options.reduce((accum, cur) => ({ ...accum, [cur.id]: cur.text }), {} as Record<string, string>);
+const statusTransitions = {
+  standBy: { id: 0, newStatus: ResponderStatus.Standby, text: 'Stand By' },
+  standDown: { id: 1, newStatus: ResponderStatus.SignedOut, text: 'Stand Down' },
+  inTown: { id: 2, newStatus: ResponderStatus.Remote, text: 'In Town' },
+  signIn: { id: 3, newStatus: ResponderStatus.SignedIn, text: 'Sign In' },
+  turnAround: { id: 4, newStatus: ResponderStatus.Demobilized, text: 'Turn Around' },
+  arriveBase: { id: 5, newStatus: ResponderStatus.Available, text: 'Arrive Base' },
+  departBase: { id: 6, newStatus: ResponderStatus.Demobilized, text: 'Depart Base' },
+  signOut: { id: 7, newStatus: ResponderStatus.SignedOut, text: 'Sign Out' },
+  resetStatus: { id: 8, newStatus: ResponderStatus.NotResponding, text: 'Reset Status' }, // clear status in edge cases that shouldn't generally be possible.
+  assigned: { id: 9, newStatus: ResponderStatus.Assigned, text: 'Assigned' },
+  available: { id: 10, newStatus: ResponderStatus.Available, text: 'Available' }
+}
 
-function getRecommendedAction(current: ResponderStatus|undefined, startTime: number): ResponderStatus {
-  const now = new Date().getTime();
-  if (current === ResponderStatus.SignedIn) {
-    return ResponderStatus.SignedOut;
-  }
-  if (startTime - 60 * 60 * 1000 > now) {
-    return ResponderStatus.Standby;
+const statusOptions: Record<ResponderStatus, { id: number, newStatus: ResponderStatus, text: string }[]> = {
+  [ResponderStatus.NotResponding]: [statusTransitions.signIn, statusTransitions.standBy, statusTransitions.inTown],
+  [ResponderStatus.Standby]: [statusTransitions.signIn, statusTransitions.standDown],
+  [ResponderStatus.Remote]: [statusTransitions.signOut],
+  [ResponderStatus.SignedIn]: [statusTransitions.arriveBase, statusTransitions.turnAround, statusTransitions.signOut],
+  [ResponderStatus.Available]: [statusTransitions.departBase, statusTransitions.assigned],
+  [ResponderStatus.Assigned]: [statusTransitions.available],
+  [ResponderStatus.Demobilized]: [statusTransitions.signOut, statusTransitions.signIn, statusTransitions.arriveBase],
+  [ResponderStatus.SignedOut]: [statusTransitions.signIn, statusTransitions.standBy, statusTransitions.inTown],
+}
+
+const futureOptions: Record<ResponderStatus, { id: number, newStatus: ResponderStatus, text: string }[]> = {
+  [ResponderStatus.NotResponding]: [statusTransitions.standBy],
+  [ResponderStatus.Standby]: [statusTransitions.standDown],
+  [ResponderStatus.Remote]: [statusTransitions.resetStatus],
+  [ResponderStatus.SignedIn]: [statusTransitions.resetStatus],
+  [ResponderStatus.Available]: [statusTransitions.resetStatus],
+  [ResponderStatus.Assigned]: [statusTransitions.resetStatus],
+  [ResponderStatus.Demobilized]: [statusTransitions.resetStatus],
+  [ResponderStatus.SignedOut]: [statusTransitions.standBy],
+}
+
+const fourHourEarlySigninWindow = 4 * 60 * 60 * 1000;

```

Comment:
nit: I'm OK being overridden on this, but my preference would be for this variable to be named "earlySignInWindow" and a comment notate the length.

If the value ever needs to be changed in the future, only one line needs to be updated rather than 2.

Reply:
Done